Xamarin.ExposureNotification/ExposureNotification.ios.cs

```diff
@@ -66,117 +62,117 @@ static async Task<bool> PlatformIsEnabled()
 			return m.ExposureNotificationEnabled;
 		}
 
-		// Gets the contact info of anyone the user had contact with who was diagnosed
-		static async Task<IEnumerable<ExposureInfo>> PlatformGetExposureInformation()
-		{
-			var s = await GetSessionAsync();
-
-			var exposures = new List<ENExposureInfo>();
-
-			ENExposureDetectionSessionGetExposureInfoResult result;
-			do
-			{
-				result = await s.GetExposureInfoAsync(100);
-				exposures.AddRange(result.Exposures);
-			} while (!result.Done);
-
-			return exposures.Select(i =>
-				new ExposureInfo(
-					((DateTime)i.Date).ToLocalTime(),
-					TimeSpan.FromMinutes(i.Duration),
-					i.AttenuationValue,
-					i.TotalRiskScore,
-					i.TransmissionRiskLevel.FromNative()));
-		}
-
-		// Call this when the user has confirmed diagnosis
-		static async Task PlatformSubmitSelfDiagnosis()
-		{
-			var m = await GetManagerAsync();
-			var selfKeys = await m.GetDiagnosisKeysAsync();
-
-			await Handler.UploadSelfExposureKeysToServer(
-				selfKeys.Select(k => new TemporaryExposureKey(
-					k.KeyData.ToArray(),
-					k.RollingStartNumber,
-					TimeSpan.FromMinutes(k.RollingStartNumber),
-					k.TransmissionRiskLevel.FromNative())));
-		}
-
 		// Tells the local API when new diagnosis keys have been obtained from the server
-		static async Task PlatformAddDiagnosisKeys(IEnumerable<TemporaryExposureKey> diagnosisKeys)
+		static async Task<(ExposureDetectionSummary, IEnumerable<ExposureInfo>)> PlatformDetectExposuresAsync(IEnumerable<TemporaryExposureKey> diagnosisKeys)
 		{
-			var s = await GetSessionAsync();
-
-			// Batch up adding our items
-			var batchSize = (int)s.MaximumKeyCount;
-			var sequence = diagnosisKeys;
+			// Get a temporary working directory
+			var dirs = NSSearchPath.GetDirectories(NSSearchPathDirectory.CachesDirectory, NSSearchPathDomain.User).FirstOrDefault();
+			var root = Path.Combine(dirs, "diagnosisKeys");
+			if (!Directory.Exists(root))
+				Directory.CreateDirectory(root);
```

Comment:
A little messy and we might want to do this better, but create a new folder to save the temp files in.